tests/contract/test_log_emission.py

```diff
@@ -0,0 +1,80 @@
+from sharding.handler.utils.web3_utils import (
+    mine,
+)
+from tests.handler.fixtures import (  # noqa: F401
+    smc_handler,
+)
+from tests.contract.utils.common_utils import (
+    fast_forward,
+)
+from tests.contract.utils.notary_account import (
+    NotaryAccount,
+)
+from tests.contract.utils.sample_helper import (
+    sampling,
+)
+
+
+def test_log_emission(smc_handler):  # noqa: F811
+    web3 = smc_handler.web3
+    notary = NotaryAccount(0)
+
+    # Register
+    tx_hash = smc_handler.register_notary(private_key=notary.private_key)
+    mine(web3, 1)
+    # Check that log was successfully emitted
+    tx_receipt = web3.eth.getTransactionReceipt(tx_hash)
+    log = smc_handler.events.RegisterNotary().processReceipt(tx_receipt)[0]['args']
```

Comment:
Thinking if we need some APIs for those special logs (`AddHeader`, `SubmitVote`, `RegisterNotary`, `DeregisterNotary`, `ReleaseNotary` ...) in `SMCHandler`. What do you think of that?

Reply:
Yes. I was thinking adding these AIPs in `ShardTracker`.
`shard_id` in these event will be indexed so the logs emitted will become searchable(search by `shard_id`) and so clients using `ShardTracker` will get exactly the logs they care.